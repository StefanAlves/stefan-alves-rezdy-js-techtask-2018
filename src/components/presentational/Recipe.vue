<!-- src/components/Recipe.vue -->
<template>
  <div class="recipe">
    <div class="title">{{recipe.title}}</div>
    <ul class="ingredients">
      <li class="ingredient" v-for="(ingredient, index) in recipe.ingredients" :key="index">
        {{ingredient}}
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from "vue-property-decorator";

@Component
export default class Recipe extends Vue {
  @Prop()
  recipe!: string;
}
</script>

<style>
.recipe {
  border: 1px solid #eeeeee;
  border-radius: 4px;
  box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.2),
    0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 2px 1px -1px rgba(0, 0, 0, 0.12);
  font-size: 13px;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  margin: 10px 0px;
}
.title {
  border-bottom: 1px solid #eeeeee;
  background: #fafafa;
  color: rgba(0, 0, 0, 0.7);
  font-size: 17px;
  font-weight: bold;

  padding: 10px;
}
.ingredients {
  color: rgba(0, 0, 0, 0.7);
  padding: 0px;
}
.ingredient {
  list-style: none;
  padding: 5px 10px;
}
</style>
