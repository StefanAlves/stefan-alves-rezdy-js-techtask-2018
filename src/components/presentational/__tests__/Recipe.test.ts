import { mount, shallowMount } from '@vue/test-utils';
import Recipe from '../Recipe.vue';

describe('Recipe', () => {
    const recipe = {
        title: 'Fry-up',
        ingredients: [
            'Bacon',
            'Eggs',
            'Baked Beans',
            'Mushrooms',
            'Sausage',
            'Bread',
        ],
    };

    it('renders correctly', () => {
        const wrapper = mount(Recipe, {
            propsData: { recipe },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render', () => {

        const wrapper = shallowMount(Recipe, {
            propsData: { recipe },
        });
        const div = wrapper.findAll('div').at(1);
        const li = wrapper.findAll('li');
        expect(div.text()).toEqual('Fry-up');
        expect(li.length).toEqual(6);
        expect(wrapper.props().recipe.title).toEqual('Fry-up');
        expect(wrapper.props().recipe.ingredients).toEqual([
            'Bacon',
            'Eggs',
            'Baked Beans',
            'Mushrooms',
            'Sausage',
            'Bread',
        ]);
    });
});
