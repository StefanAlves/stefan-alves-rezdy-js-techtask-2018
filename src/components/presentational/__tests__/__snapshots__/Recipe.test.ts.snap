// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe renders correctly 1`] = `
<div
  class="recipe"
>
  <div
    class="title"
  >
    Fry-up
  </div>
   
  <ul
    class="ingredients"
  >
    <li
      class="ingredient"
    >
      
      Bacon
    
    </li>
    <li
      class="ingredient"
    >
      
      Eggs
    
    </li>
    <li
      class="ingredient"
    >
      
      Baked Beans
    
    </li>
    <li
      class="ingredient"
    >
      
      Mushrooms
    
    </li>
    <li
      class="ingredient"
    >
      
      Sausage
    
    </li>
    <li
      class="ingredient"
    >
      
      Bread
    
    </li>
  </ul>
</div>
`;
